Habitica/src/main/java/com/habitrpg/android/habitica/ui/activities/CreateChallengeActivity.java

```diff
@@ -0,0 +1,563 @@
+        if (supportActionBar != null) {
+            supportActionBar.setDisplayShowHomeEnabled(true);
+            supportActionBar.setHomeAsUpIndicator(R.drawable.ic_close_white_24dp);
+
+            supportActionBar.setTitle("");
+            supportActionBar.setBackgroundDrawable(new ColorDrawable(resources.getColor(R.color.brand_200)));
+            supportActionBar.setElevation(0);
+        }
+
+        locationAdapter = new GroupArrayAdapter(this);
+        locationAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        challengeLocalRepository.getGroups().subscribe(groups -> {
+            Group tavern = new Group();
+            tavern.id = "00000000-0000-4000-A000-000000000000";
+            tavern.name = getString(R.string.tavern);
+
+            locationAdapter.add(tavern);
+
+            groups.forEach(group -> locationAdapter.add(group));
+        }, Throwable::printStackTrace);
+
+        challenge_location_spinner.setAdapter(locationAdapter);
+        challenge_location_spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                checkGemAndLocation();
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+        create_challenge_prize.setOnKeyListener((view, i, keyEvent) -> {
+            checkGemAndLocation();
+
+            return false;
+        });
+
+        addHabit = createTask(ChallengeTasksRecyclerViewAdapter.TASK_TYPE_ADD_ITEM, resources.getString(R.string.add_habit));
+        addDaily = createTask(ChallengeTasksRecyclerViewAdapter.TASK_TYPE_ADD_ITEM, resources.getString(R.string.add_daily));
+        addTodo = createTask(ChallengeTasksRecyclerViewAdapter.TASK_TYPE_ADD_ITEM, resources.getString(R.string.add_todo));
+        addReward = createTask(ChallengeTasksRecyclerViewAdapter.TASK_TYPE_ADD_ITEM, resources.getString(R.string.add_reward));
+
+
+        ArrayList<Task> taskList = new ArrayList<>();
+        taskList.add(addHabit);
+        taskList.add(addDaily);
+        taskList.add(addTodo);
+        taskList.add(addReward);
+
+        challengeTasks = new ChallengeTasksRecyclerViewAdapter(null, 0, this, "", null, false, true);
+        challengeTasks.setTasks(taskList);
+        challengeTasks.enableAddItem(t -> {
+            if (t.equals(addHabit)) {
+                openNewTaskActivity(Task.TYPE_HABIT);
+            } else if (t.equals(addDaily)) {
+                openNewTaskActivity(Task.TYPE_DAILY);
+            } else if (t.equals(addTodo)) {
+                openNewTaskActivity(Task.TYPE_TODO);
+            } else if (t.equals(addReward)) {
+                openNewTaskActivity(Task.TYPE_REWARD);
+            }
+        });
+
+        create_challenge_task_list.addOnItemTouchListener(new RecyclerView.SimpleOnItemTouchListener() {
```

Comment:
Hmm I could try to implement `ListAdapter` in to `ChallengeTasksRecyclerViewAdapter` because a ListView won't accept `RecyclerView.Adapter`